schema/schemaspec/schemahcl/hcl_test.go

```diff
@@ -143,3 +144,49 @@ func ExampleMarshal() {
 	//   y = 1
 	// }
 }
+
+func TestInterface(t *testing.T) {
+	f := `
+zoo "ramat_gan" {
+	lion "simba" {
+		friend = "rafiki"
+	}
+	parrot "iago" {
+		boss = "jafar"
+	}
+}
+`
+	type (
+		Animal interface {
+			animal()
+		}
+		Parrot struct {
+			Animal
+			Boss string `spec:"boss"`
+		}
+		Lion struct {
+			Animal
+			Friend string `spec:"friend"`
+		}
+		Zoo struct {
+			Animals []Animal `spec:""`
+		}
```

Comment:
Can you add a test case for something like:

```go
type U struct {
	Pet Animal `spec:""`
}
```

Reply:
done